src/modules/auth/auth.service.ts

```diff
@@ -117,4 +117,30 @@ export class AuthService {
     });
     return authToken;
   }
+
+  async googleLogin(user: User) {
+    const { googleId, email, firstName, lastName, picture } = user;
+
+    const existingUser = await this.findByGoogleId(googleId);
+
+    if (!existingUser) {
+      const newUser = this.userRepository.create({
+        googleId,
+        email,
+        firstName,
+        lastName,
+        picture,
+      });
+      await this.userRepository.save(newUser);
+    }
+    return {
+      message: 'User info From Google',
+      user,
+    };
+  }
+
+  async findByGoogleId(id: string) {
+    const user = await this.userRepository.findOne({ where: { googleId: id } });
```

Comment:
Just a quick note for future: we can use ```findOneBy()``` if conditions are simple. For example, ```findOneBy({ googleId: id })``` in this case. Makes code a little cleaner.